core/run.go

```diff
@@ -177,6 +178,7 @@ func Run(
 	indexer := query.NewIndexer(db, c, pinStore)
 
 	a := &API{
+		ctx:          ctx,
```

Comment:
I'd really rather avoid this. I see it's just for plumbing to get the context in to fetch.New, but it would be too easy to reuse this field elsewhere. There's a suggestion above for how we might avoid this, but if none of those are viable, at least can we set it to nil after applying the options?